client/src/templates/Challenges/classic/editor.tsx

```diff
@@ -409,7 +409,59 @@ const Editor = (props: EditorProps): JSX.Element => {
         });
       }
     });
+    editor.addAction({
+      id: 'toggle-aria-roledescription',
+      label: 'Toggle JAWS 2022 editor bugfix',
+      keybindings: [
+        monaco.KeyMod.CtrlCmd | monaco.KeyMod.Alt | monaco.KeyCode.KEY_J
+      ],
+      run: toggleAriaRoledescription
+    });
     editor.onDidFocusEditorWidget(() => props.setEditorFocusability(true));
+
+    // aria-roledescription is set by default, check if it needs
+    // to be removed.
+    const ariaRoledescription = getStoredAriaRoledescription();
+    if (!ariaRoledescription) {
+      setAriaRoledescription(false);
+    }
+  };
+
+  const toggleAriaRoledescription = () => {
+    const newValue = !getStoredAriaRoledescription();
+    setAriaRoledescription(newValue);
+    ariaAlert(`JAWS 2022 editor bugfix turned ${!newValue ? 'on' : 'off'}`);
```

Comment:
I'm wondering if we want to rework this bit.

The bug fix is to remove the roledescription, right? So currently if `newValue` is set to `true`, then the roledescription is set, which means the bug is not fixed.

This code works, but the logic seems a bit counterintuitive for the alert string and took me a bit to puzzle out.

Reply:
Yes, I agree that this is a little counterintuitive. The state of the attribute is stored as:

`true`: `aria-roledescription` is set to "editor" on textarea (this is the default state)
`false`: `aria-roledescription` is removed from textarea (this is the bugfix)

When we toggle the state to `false` then the bug is "on", and vice-versa.

I think the issue here is that even though this is being added to address a very specific bug in JAWS, the underlying functionality is just toggling an attribute, which is reflected in the code, but I am trying to hide these gruesome technical details from the user and refer to it as the "JAWS bugfix". The disconnect comes because when the attribute is `false` the bugfix is "on". 

I'm not sure that the alert message (and menu label) should actually refer to the JAWS bugfix. Perhaps they should just refer to the status of the attribute? The menu label would be something like "Toggle roledescription attribute" and the announcement after the toggle would be something like "roledescription turned (off|on)". And then we could have instructions for people affected by this bug that explains what to do.

Reply:
I think that's a more explicit approach, yea. And will ideally help us avoid total confusion when we look at this code in six months. 😁

Reply:
This has been updated.